подключить Гипернайт

beerbis opened the pull request.

Commit:
feat: подключить hibernate и начать хранить Product-ы в базе

beerbis commented on boot/src/main/java/ru/beerbis/springer/entity/Product.java:
```diff
@@ -2,26 +2,41 @@
-        this.cost = requireNonNull(cost, "cost");
-
-        if (cost < 0) {
-            throw new IllegalArgumentException("cost may noy be negative, but: cost=" + cost);
```
очень неприятный момент, что теперь фактически никакого контроля целостности нет

Lezenford replied:
Вы можете встроить эту проверку в сеттер, например. Либо прикрутить валидацию отдельно - есть механизмы. Но сам гибернейт таких не предоставляет

Commit:
fix: устранить чтение фантомов из "контекста персистентности", заиспользовать D из CRUD entityManager-а

Кеширование стало неожиданностью. Удалил прямым запросом, а редактирование продолжает показывать позицию.
Варианты: либо перед чтением из контекст сбрасывать его, либо вот так удалять с предварительным чтением...

Commit:
fix: учесть рекомендации по работе с hibernate из других "лекций"

Иные лекции рекомендуют держать с сессиии(entityManager) лишь в рамках выполнения конкретной операции,
и точно никак не на протяжении всего времени выполнения сервиса, ровно по причине того, что сессии - это
"конекст постоянсва"/кеш, и раздувать его нет никакого смысла, как там говорится, а то и вредно. Раз мы не храним
кеш между обращениями - можно вернуть удаление одним вызовом, что, кстати, так же рекомендуется в иных
лекциях - удалять отдельным запросом, единичным обращением.

перешёл на использование сессий гиберната вместо EntityManager-а, т.к. те `Autoclosable`, что удобно, благодаря try-c-ресурсами.
Так же такой переход решает проблему черезчур рачительного обихаживания "транзакций" "контекста постоянства": закрылся контекст - вот и хорошо,
если хотели сохранить/провести какие-то изменения через CRUD - выполняем `Commit` , а если нет - можно просто ничего не делать, и всё. Красота.

Обо всём этом я оставлял комментарии ранее, но.. видимо они малозначительны.
